File: Semana_2/Playwright/javascript/npm/curso/e2e/pages/classic/login.page.js
const { AccountsOverviewPage } = require('./accounts-overview.page');
const { capture } = require('../evidence');

const URL = 'https://parabank.parasoft.com/parabank/index.htm';

/**
 * POM clasico: localizadores y acciones viven juntos.
 */
class LoginPage {
  constructor(page) {
    this.page = page;
    // ParaBank no asocia labels accesibles a estos campos, por eso aqui CSS sigue siendo la opcion clara.
    this.usernameInput = page.locator('input[name="username"]');
    this.passwordInput = page.locator('input[name="password"]');
    this.loginButton = page.getByRole('button', { name: 'Log In' });
  }

  async open() {
    await this.page.goto(URL);
    await capture(this.page, 'parabank-open-login');
    return this;
  }

  async loginAs(username, password) {
    await this.usernameInput.fill(username);
    await capture(this.page, 'parabank-type-username');
    await this.passwordInput.fill(password);
    await capture(this.page, 'parabank-type-password');
    await this.loginButton.click();
    await capture(this.page, 'parabank-click-login');
    return new AccountsOverviewPage(this.page);
  }
}

module.exports = { LoginPage };

File: Semana_2/Playwright/Java/maven/curso/e2e/src/test/java/com/curso/playwright/pages/classic/LoginPage.java
package com.curso.playwright.pages.classic;

import com.curso.playwright.evidence.EvidenceSupport;
import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

/**
 * POM clasico: localizadores y acciones viven juntos.
 */
public class LoginPage {
    private static final String URL = "https://parabank.parasoft.com/parabank/index.htm";

    private final Page page;
    private final Locator usernameInput;
    private final Locator passwordInput;
    private final Locator loginButton;

    public LoginPage(Page page) {
        this.page = page;
        // ParaBank no asocia labels accesibles a estos campos, por eso aqui CSS sigue siendo la opcion clara.
        this.usernameInput = page.locator("input[name='username']");
        this.passwordInput = page.locator("input[name='password']");
        this.loginButton = page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Log In"));
    }

    public LoginPage open() {
        page.navigate(URL);
        EvidenceSupport.capture(page, "parabank-open-login");
        return this;
    }

    public AccountsOverviewPage loginAs(String username, String password) {
        usernameInput.fill(username);
        EvidenceSupport.capture(page, "parabank-type-username");
        passwordInput.fill(password);
        EvidenceSupport.capture(page, "parabank-type-password");
        loginButton.click();
        EvidenceSupport.capture(page, "parabank-click-login");
        return new AccountsOverviewPage(page);
    }
}
